specs/MultiProfile.md

```diff
@@ -0,0 +1,274 @@
+API spec for multiple profile support
+===
+
+# Background
+
+Previously, all WebView2s can only use one fixed Edge profile in the browser process, which is
+normally the **Default** profile or the profile specified by the **--profile-directory** command
+line switch. It means different WebView2s share a single profile directory on disk for data storage,
+which might bring security concerns over cookies, autofill data, and password management etc.. Also,
+they might also interfere with each other in terms of user preference settings.
+
+Although you can make your WebView2s use different user data directories to achieve data separation,
+in such way you'll have to be running multiple browser instances (each including a browser process
+and a bunch of child processes), which means much more consumption for system resources including
+memory, CPU footprint, disk space etc. so it is not desirable.
+
+With all above, we're adding these new APIs to support multiple profiles, so that you can have
+multiple WebView2s running with separate profiles under a single user data directory (i.e. a single
+browser instance at runtime), which means separate cookies, user preference settings, and various
+data storage etc., to help you build a more wonderful experience for your application.
+
+# Examples
+
+## Win32 C++
+
+### Provide options to create WebView2 with a specific profile
+
+```cpp
+HRESULT AppWindow::CreateControllerWithOptions()
+{
+    auto webViewEnvironment4 =
+        m_webViewEnvironment.try_query<ICoreWebView2Environment4>();
+    if (!webViewEnvironment4)
+    {
+        FeatureNotAvailable();
+        return S_OK;
+    }
+
+    Microsoft::WRL::ComPtr<ICoreWebView2ControllerOptions> options;
+    HRESULT hr = webViewEnvironment4->CreateCoreWebView2ControllerOptions(
+        m_webviewOption.profile.c_str(), m_webviewOption.isInPrivate, options.GetAddressOf());
+    if (hr == E_INVALIDARG)
```

Comment:
What if the profile name already exists? Error, or it gets re-used?

Reply:
Correct, reused. Must document

Reply:
Done. I added it when calling `put_ProfileName` function. Thanks.